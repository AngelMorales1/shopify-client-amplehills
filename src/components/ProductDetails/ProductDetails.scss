@import 'settings';

.ProductDetails {
  &--Blue {
    @extend .bg-sky-blue;
  }
}
