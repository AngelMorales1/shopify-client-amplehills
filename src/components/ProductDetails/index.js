import React, { Component } from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import cx from 'classnames';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';

import { Image, Button } from 'components/base';
import styles from './ProductDetails.scss';

class ProductDetails extends Component {
  constructor() {
    super(...arguments);

    this.state = {
      activeFlavor: false
    };
  }

  isActiveFlavor(id, index) {
    return (
      this.state.activeFlavor === id || (!index && !this.state.activeFlavor)
    );
  }

  render() {
    const { block, z, setRef } = this.props;
    const fields = get(block, 'fields', {});
    const details = get(fields, 'productDetails', []);
    const colorClass = `ProductDetails--${get(fields, 'color', '')}`;

    return (
      <div
        ref={refBlock => setRef(refBlock)}
        className={cx(styles['ProductDetails'], styles[colorClass], 'p2 drip')}
        style={{ zIndex: z }}
      >
        <div className="flex justify-center flex-wrap center mb3">
          <h2 className="block-headline w100 my3">The Details</h2>
          {details.map((detail, i) => {
            const color = this.isActiveFlavor(get(detail, 'sys.id', ''), i)
              ? 'clear-madison-blue-border'
              : 'madison-blue';

            return (
              <Button
                className="m1"
                color={color}
                variant="primary-small"
                shadow={true}
                key={get(detail, 'sys.id', i)}
                label={get(detail, 'fields.title', '')}
                onClick={() =>
                  this.setState({
                    activeFlavor: get(detail, 'sys.id', '')
                  })
                }
              />
            );
          })}
        </div>
        <div className={`${styles['ProductDetails--container']}`}>
          {details.map((detail, i) => {
            const fields = get(detail, 'fields', {});
            const classes = cx(
              styles['ProductDetail'],
              'transition-slide-up-large container-width mx-auto flex items-center py3',
              {
                [styles['ProductDetail--active']]: this.isActiveFlavor(
                  get(detail, 'sys.id', ''),
                  i
                )
              }
            );

            return (
              <div className={classes} key={get(detail, 'sys.id', '')}>
                <div className="col-12 md-col-6">
                  <div
                    className={cx(
                      styles['ProductDetail--description'],
                      'flex justify-center items-center mb4'
                    )}
                  >
                    <div
                      className={cx(
                        styles['ProductDetail--description-image'],
                        'col-3 md-col-2'
                      )}
                    >
                      <Image
                        alt={`${get(fields, 'title', '')} description image`}
                        src={contentfulImgUtil(
                          get(fields, 'text1Image.fields.file.url', ''),
                          '500',
                          'png'
                        )}
                      />
                    </div>
                    <div className="md-col-10">
                      <p
                        dangerouslySetInnerHTML={{
                          __html: marked(get(fields, 'text1', ''))
                        }}
                        className={cx(
                          styles['ProductDetail--description-text'],
                          'markdown-block'
                        )}
                      />
                    </div>
                  </div>
                  <div
                    className={cx(
                      styles['ProductDetail--description'],
                      'flex items-center'
                    )}
                  >
                    <div
                      className={cx(
                        styles['ProductDetail--description-image'],
                        'col-3 md-col-2'
                      )}
                    >
                      <Image
                        alt={`${get(fields, 'title', '')} description image`}
                        src={contentfulImgUtil(
                          get(fields, 'text2Image.fields.file.url', ''),
                          '500',
                          'png'
                        )}
                      />
                    </div>
                    <div className="md-col-10">
                      <div
                        dangerouslySetInnerHTML={{
                          __html: marked(get(fields, 'text2', ''))
                        }}
                        className={cx(
                          styles['ProductDetail--description-text'],
                          'markdown-block'
                        )}
                      />
                    </div>
                  </div>
                </div>
                <div
                  className={`${
                    styles['FlavorHighlight']
                  } relative col-10 md-col-5 z-1`}
                >
                  <div
                    className={`${
                      styles['FlavorHighlight--wrapper']
                    } z-1 absolute flex justify-center items-center`}
                  >
                    <p
                      className={cx(
                        styles['FlavorHighlight--label-text'],
                        'absolute center callout-small z-1'
                      )}
                    >
                      {get(fields, 'flavorHighlight', '')}
                    </p>
                    <div
                      className={`${
                        styles['FlavorHighlight--label']
                      } p3 circle square bg-yellow`}
                    />
                  </div>
                  <div>
                    <div
                      className="circle square"
                      style={{
                        background: `url(${contentfulImgUtil(
                          get(
                            fields,
                            'flavorHighlightImage.fields.file.url',
                            ''
                          ),
                          '1400'
                        )}) no-repeat center`,
                        backgroundSize: 'cover'
                      }}
                    />
                  </div>
                </div>
              </div>
            );
          })}
        </div>
      </div>
    );
  }
}

ProductDetails.propTypes = {
  z: PropTypes.number,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      color: PropTypes.string,
      productDetails: PropTypes.arrayOf(
        PropTypes.shape({
          fields: PropTypes.shape({
            title: PropTypes.string,
            flavorHighlight: PropTypes.string,
            flavorHighlightImage: PropTypes.shape({
              fields: PropTypes.shape({
                description: PropTypes.string,
                file: PropTypes.shape({
                  url: PropTypes.string
                })
              })
            }),
            text1: PropTypes.string,
            text1Image: PropTypes.shape({
              fields: PropTypes.shape({
                description: PropTypes.string,
                file: PropTypes.shape({
                  url: PropTypes.string
                })
              })
            }),
            text2: PropTypes.string,
            text2Image: PropTypes.shape({
              fields: PropTypes.shape({
                description: PropTypes.string,
                file: PropTypes.shape({
                  url: PropTypes.string
                })
              })
            })
          }),
          sys: PropTypes.shape({
            id: PropTypes.string
          })
        })
      )
    })
  }),
  setRef: PropTypes.func
};

ProductDetails.defaultProps = {
  z: 1,
  block: {
    fields: {
      color: 'Blue',
      productDetails: [
        {
          fields: {
            title: '',
            flavorHighlight: '',
            flavorHighlightImage: {
              fields: {
                description: '',
                file: {
                  url: ''
                }
              }
            }
          },
          sys: {
            id: ''
          }
        }
      ]
    }
  },
  setRef: () => {}
};

export default ProductDetails;
