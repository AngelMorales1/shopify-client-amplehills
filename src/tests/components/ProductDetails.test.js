import React from 'react';
import { shallow } from 'enzyme';

import ProductDetails from 'components/ProductDetails';
import contentful from './data/ProductDetails.contentful.js';

it('renders without data', () => {
  const component = shallow(<ProductDetails />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(<ProductDetails data={contentful} z={1} />);

  expect(component).toMatchSnapshot();
});
