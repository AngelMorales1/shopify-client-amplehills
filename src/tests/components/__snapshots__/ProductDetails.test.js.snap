// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductDetails
    block={Object {}}
    data={
        Object {
            "color": "Blue",
            "productDetails": Array [
              Object {
                "fields": Object {
                  "description": "Inspired by the harmony and peaceful nature of the Jedi, The Force illustrates the precarious balance between the dark and light side with sweet cream ice cream, swirled with rich chocolate fudge, and set against a galaxy of white and dark chocolate pearls.",
                  "details": "Housemade munchies mix-in",
                  "detailsImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 258,
                            "width": 234,
                          },
                          "size": 84469,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.34 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/1U89EfHNAcYGgoW6yM6qKA/9d8e1bdb12caa5283c48a466f88ca8e0/Screen_Shot_2018-06-07_at_5.57.34_PM.png",
                      },
                      "title": "PB Munchies detail photo",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T22:00:03.109Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "1U89EfHNAcYGgoW6yM6qKA",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T22:00:03.109Z",
                    },
                  },
                  "flavorHighlight": "Blended rainbow splendor yay!",
                  "flavorHighlightImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 730,
                            "width": 654,
                          },
                          "size": 763798,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.41 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/2C9hEcZpk4IWwMU4eqakke/12d17e080d490bc9f0f30b0396ba3f7e/Screen_Shot_2018-06-07_at_5.57.41_PM.png",
                      },
                      "title": "PB Munchies flavor highlight",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T22:01:02.371Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "2C9hEcZpk4IWwMU4eqakke",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T22:01:02.371Z",
                    },
                  },
                  "pintImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 282,
                            "width": 232,
                          },
                          "size": 89251,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.30 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/3EokEYLPio8EKKS8K6eeOW/70e220695d2e8c29075a56737261cb63/Screen_Shot_2018-06-07_at_5.57.30_PM.png",
                      },
                      "title": "PB Munchies Pint",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T21:59:16.349Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "3EokEYLPio8EKKS8K6eeOW",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T21:59:16.349Z",
                    },
                  },
                  "title": "The Munchies",
                },
                "sys": Object {
                  "contentType": Object {
                    "sys": Object {
                      "id": "productDetails",
                      "linkType": "ContentType",
                      "type": "Link",
                    },
                  },
                  "createdAt": "2018-06-07T22:01:09.471Z",
                  "environment": Object {
                    "sys": Object {
                      "id": "master",
                      "linkType": "Environment",
                      "type": "Link",
                    },
                  },
                  "id": "2qxLEa6moYEOGqgem0yomy",
                  "locale": "en-US",
                  "revision": 3,
                  "space": Object {
                    "sys": Object {
                      "id": "8askpxhbyg64",
                      "linkType": "Space",
                      "type": "Link",
                    },
                  },
                  "type": "Entry",
                  "updatedAt": "2018-06-12T19:05:22.730Z",
                },
              },
              Object {
                "fields": Object {
                  "description": "Inspired by the harmony and peaceful nature of the Jedi, The Force illustrates the precarious balance between the dark and light side with sweet cream ice cream, swirled with rich chocolate fudge, and set against a galaxy of white and dark chocolate pearls.",
                  "details": "Delicious bites!",
                  "detailsImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 258,
                            "width": 234,
                          },
                          "size": 84469,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.34 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/1U89EfHNAcYGgoW6yM6qKA/9d8e1bdb12caa5283c48a466f88ca8e0/Screen_Shot_2018-06-07_at_5.57.34_PM.png",
                      },
                      "title": "PB Munchies detail photo",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T22:00:03.109Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "1U89EfHNAcYGgoW6yM6qKA",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T22:00:03.109Z",
                    },
                  },
                  "flavorHighlight": "Mint green!",
                  "flavorHighlightImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 730,
                            "width": 654,
                          },
                          "size": 763798,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.41 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/2C9hEcZpk4IWwMU4eqakke/12d17e080d490bc9f0f30b0396ba3f7e/Screen_Shot_2018-06-07_at_5.57.41_PM.png",
                      },
                      "title": "PB Munchies flavor highlight",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T22:01:02.371Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "2C9hEcZpk4IWwMU4eqakke",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T22:01:02.371Z",
                    },
                  },
                  "pintImage": Object {
                    "fields": Object {
                      "file": Object {
                        "contentType": "image/png",
                        "details": Object {
                          "image": Object {
                            "height": 282,
                            "width": 232,
                          },
                          "size": 89251,
                        },
                        "fileName": "Screen Shot 2018-06-07 at 5.57.30 PM.png",
                        "url": "//images.ctfassets.net/8askpxhbyg64/3EokEYLPio8EKKS8K6eeOW/70e220695d2e8c29075a56737261cb63/Screen_Shot_2018-06-07_at_5.57.30_PM.png",
                      },
                      "title": "PB Munchies Pint",
                    },
                    "sys": Object {
                      "createdAt": "2018-06-07T21:59:16.349Z",
                      "environment": Object {
                        "sys": Object {
                          "id": "master",
                          "linkType": "Environment",
                          "type": "Link",
                        },
                      },
                      "id": "3EokEYLPio8EKKS8K6eeOW",
                      "locale": "en-US",
                      "revision": 1,
                      "space": Object {
                        "sys": Object {
                          "id": "8askpxhbyg64",
                          "linkType": "Space",
                          "type": "Link",
                        },
                      },
                      "type": "Asset",
                      "updatedAt": "2018-06-07T21:59:16.349Z",
                    },
                  },
                  "title": "Peppermint Patty",
                },
                "sys": Object {
                  "contentType": Object {
                    "sys": Object {
                      "id": "productDetails",
                      "linkType": "ContentType",
                      "type": "Link",
                    },
                  },
                  "createdAt": "2018-06-07T22:04:41.871Z",
                  "environment": Object {
                    "sys": Object {
                      "id": "master",
                      "linkType": "Environment",
                      "type": "Link",
                    },
                  },
                  "id": "6HC1913RW806I4cmESEY8C",
                  "locale": "en-US",
                  "revision": 2,
                  "space": Object {
                    "sys": Object {
                      "id": "8askpxhbyg64",
                      "linkType": "Space",
                      "type": "Link",
                    },
                  },
                  "type": "Entry",
                  "updatedAt": "2018-06-12T19:05:31.068Z",
                },
              },
            ],
            "title": "Our Classics 4 Pack",
          }
    }
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex justify-center flex-wrap center mb3"
>
          <h2
                    className="block-headline w100 my3"
          >
                    The Details
          </h2>
</div>,
        <div
          className="undefined"
>
          
</div>,
      ],
      "className": "p2 drip",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline w100 my3"
>
              The Details
</h2>,
            Array [],
          ],
          "className": "flex justify-center flex-wrap center mb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Details",
              "className": "block-headline w100 my3",
            },
            "ref": null,
            "rendered": "The Details",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "undefined",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex justify-center flex-wrap center mb3"
>
            <h2
                        className="block-headline w100 my3"
            >
                        The Details
            </h2>
</div>,
          <div
            className="undefined"
>
            
</div>,
        ],
        "className": "p2 drip",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline w100 my3"
>
                The Details
</h2>,
              Array [],
            ],
            "className": "flex justify-center flex-wrap center mb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The Details",
                "className": "block-headline w100 my3",
              },
              "ref": null,
              "rendered": "The Details",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "undefined",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductDetails
    block={Object {}}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex justify-center flex-wrap center mb3"
>
          <h2
                    className="block-headline w100 my3"
          >
                    The Details
          </h2>
</div>,
        <div
          className="undefined"
>
          
</div>,
      ],
      "className": "p2 drip",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline w100 my3"
>
              The Details
</h2>,
            Array [],
          ],
          "className": "flex justify-center flex-wrap center mb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Details",
              "className": "block-headline w100 my3",
            },
            "ref": null,
            "rendered": "The Details",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "undefined",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex justify-center flex-wrap center mb3"
>
            <h2
                        className="block-headline w100 my3"
            >
                        The Details
            </h2>
</div>,
          <div
            className="undefined"
>
            
</div>,
        ],
        "className": "p2 drip",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline w100 my3"
>
                The Details
</h2>,
              Array [],
            ],
            "className": "flex justify-center flex-wrap center mb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The Details",
                "className": "block-headline w100 my3",
              },
              "ref": null,
              "rendered": "The Details",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "undefined",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
